src/update-process-env.js

```diff
@@ -0,0 +1,49 @@
+/** @babel */
+
+import {spawnSync} from 'child_process'
+
+const ENVIRONMENT_VARIABLES_TO_PRESERVE = new Set(['NODE_ENV', 'NODE_PATH'])
+
+export default function updateProcessEnv (launchEnv) {
+  let envToAssign
+  if (launchEnv.PWD) {
+    envToAssign = launchEnv
+  } else {
+    if (process.platform === 'darwin') {
+      envToAssign = getEnvFromShell()
+    }
+  }
+
+  if (envToAssign) {
+    for (let key in process.env) {
+      if (!ENVIRONMENT_VARIABLES_TO_PRESERVE.has(key)) {
+        delete process.env[key]
+      }
+    }
+
+    for (let key in envToAssign) {
+      if (!ENVIRONMENT_VARIABLES_TO_PRESERVE.has(key)) {
+        process.env[key] = envToAssign[key]
+      }
+    }
+  }
+}
+
+function getEnvFromShell () {
```

Comment:
:-1: on the basis that it's not worth taking on additional dependencies to replace this small piece of code (which has since evolved to be more simple and address more shells and more platforms).

I write a lot of go code, so to draw inspiration from one of the creators of go:

> "A little copying is better than a little dependency" 
> - [Rob Pike](http://go-proverbs.github.io)

This is cogent, given parables like `left-pad` (not to suggest consistent-env is in any way like left-pad).